packages/block-editor/src/components/heading-level-control/index.js

```diff
@@ -48,9 +54,9 @@ const POPOVER_PROPS = {
 export default function HeadingLevelControl( {
 	value,
 	onChange,
-	isParagraphAllowed = false,
+	options: { addParagraphLevel = false } = {},
```

Comment:
>I thought initially this added something like "Paragraph" to the available options but it doesn't seem like it

It does adds this option with the convention that `0` level is for paragraph. This is currently used for `Site Title` and is a bit confusing TBH and I'm not sure about the initial decision for this.

`allow empty` is not enough because we explicitly handle this as paragraph. One way to go would be to add an `additionalOptions` prop which would be something like:
```
[ { level: 0, icon: paragraph, label: 'paragraph' } ]
```
etc... but indeed being so flexible doesn't fit well in a `HeadingLevelControl`.

Another alternative is to have a similar component but not the same one for `Site Title` as it was before or even reconsider whether having `Site Title` as paragraph makes sense.

Reply:
> allow empty is not enough because we explicitly handle this as paragraph. One way to go would be to add an additionalOptions prop which would be something like:
> [ { level: 0, icon: paragraph, label: 'paragraph' } ]

This is actually exactly my initial suggestion `additionalOptions` or maybe `extraLevels` or something like that.